<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"><head><title>IBase__Exception (infer.IBase__Exception)</title><link rel="stylesheet" href="../../odoc.css"/><meta charset="utf-8"/><meta name="generator" content="odoc 1.5.3"/><meta name="viewport" content="width=device-width,initial-scale=1.0"/><script src="../../highlight.pack.js"></script><script>hljs.initHighlightingOnLoad();</script></head><body><div class="content"><header><nav><a href="../index.html">Up</a> â€“ <a href="../index.html">infer</a> &#x00BB; IBase__Exception</nav><h1>Module <code>IBase__Exception</code></h1><p>The restart scheduler and biabduction use exceptions for control flow (restarts/timeouts respectively). Functions here abstract away the semantics of when an exception can be ignored.</p></header><dl><dt class="spec type" id="type-failure_kind"><a href="#type-failure_kind" class="anchor"></a><code><span class="keyword">type</span> failure_kind</code><code> = </code><table class="variant"><tr id="type-failure_kind.FKtimeout" class="anchored"><td class="def constructor"><a href="#type-failure_kind.FKtimeout" class="anchor"></a><code>| </code><code><span class="constructor">FKtimeout</span></code></td><td class="doc"><p>max time exceeded</p></td></tr><tr id="type-failure_kind.FKsymops_timeout" class="anchored"><td class="def constructor"><a href="#type-failure_kind.FKsymops_timeout" class="anchor"></a><code>| </code><code><span class="constructor">FKsymops_timeout</span> <span class="keyword">of</span> int</code></td><td class="doc"><p>max symop's exceeded</p></td></tr><tr id="type-failure_kind.FKcrash" class="anchored"><td class="def constructor"><a href="#type-failure_kind.FKcrash" class="anchor"></a><code>| </code><code><span class="constructor">FKcrash</span> <span class="keyword">of</span> string</code></td><td class="doc"><p>uncaught exception or failed assertion</p></td></tr></table></dt><dd><p>types of biabduction failure due to timeouts</p></dd></dl><dl><dt class="spec value" id="val-pp_failure_kind"><a href="#val-pp_failure_kind" class="anchor"></a><code><span class="keyword">val</span> pp_failure_kind : Stdlib.Format.formatter <span>&#45;&gt;</span> <a href="index.html#type-failure_kind">failure_kind</a> <span>&#45;&gt;</span> unit</code></dt></dl><dl><dt class="spec exception" id="exception-Analysis_failure_exe"><a href="#exception-Analysis_failure_exe" class="anchor"></a><code><span class="keyword">exception</span> </code><code><span class="exception">Analysis_failure_exe</span> <span class="keyword">of</span> <a href="index.html#type-failure_kind">failure_kind</a></code></dt><dd><p>Timeout exception</p></dd></dl><dl><dt class="spec value" id="val-exn_not_failure"><a href="#val-exn_not_failure" class="anchor"></a><code><span class="keyword">val</span> exn_not_failure : exn <span>&#45;&gt;</span> bool</code></dt><dd><p>check that the exception is not a biabduction timeout or restart scheduler exception</p></dd></dl><dl><dt class="spec value" id="val-try_finally"><a href="#val-try_finally" class="anchor"></a><code><span class="keyword">val</span> try_finally : <span>f:<span>(unit <span>&#45;&gt;</span> <span class="type-var">'a</span>)</span></span> <span>&#45;&gt;</span> <span>finally:<span>(unit <span>&#45;&gt;</span> unit)</span></span> <span>&#45;&gt;</span> <span class="type-var">'a</span></code></dt><dd><p><code>try_finally ~f ~finally</code> executes <code>f</code> and then <code>finally</code> even if <code>f</code> raises an exception. Biabduction timeouts and restart scheduler exceptions are handled as necessary.</p></dd></dl></div></body></html>